Source/carthage/Extensions.swift

```diff
@@ -47,6 +47,38 @@ internal func print<T>(object: T) {
 	}
 }
 
+internal final class FileHandle {
+	let fileDescriptor: Int32
+	let pointer: UnsafeMutablePointer<FILE>
+
+	init(fileDescriptor fd: Int32) {
+		fileDescriptor = fd
+		pointer = fdopen(fd, "r+")
+	}
+
+	func write(data: NSData) -> Result<Int> {
+		switch Darwin.write(fileDescriptor, data.bytes, UInt(data.length)) {
```

Comment:
Yeah, I had tried debugging it for a while.

Any particular reason why Darwin's `write` is insufficient?